packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.spec.ts

```diff
@@ -241,15 +231,44 @@ function getAuthCodeRedirectHandler(
   );
 }
 
+const DEFAULT_AUTHORIZATION_CODE = "someCode";
+const DEFAULT_CODE_VERIFIER = "some code verifier";
+const DEFAULT_REDIRECT_URL = "https://some.redirect.uri";
+const DEFAULT_OAUTH_STATE = "oauth2StateValue";
+
+const mockRedirectUrl = () => {
+  const redirectUrl = new URL(DEFAULT_REDIRECT_URL);
+  redirectUrl.searchParams.append("state", DEFAULT_OAUTH_STATE);
+  redirectUrl.searchParams.append("code", DEFAULT_AUTHORIZATION_CODE);
+  return redirectUrl.href;
+};
+
+const mockDefaultStorageUtility = (
+  options: { dpop?: boolean; codeVerifier?: string; redirectUrl?: string } = {
+    dpop: true,
+    codeVerifier: DEFAULT_CODE_VERIFIER,
+    redirectUrl: DEFAULT_REDIRECT_URL,
+  }
+) =>
+  mockStorageUtility({
+    [`${USER_SESSION_PREFIX}:${DEFAULT_OAUTH_STATE}`]: {
+      sessionId: "mySession",
+    },
+    [`${USER_SESSION_PREFIX}:mySession`]: {
+      dpop: options.dpop ? "true" : "false",
+      issuer: mockIssuer().issuer.toString(),
+      codeVerifier: options.codeVerifier ?? DEFAULT_CODE_VERIFIER,
+      redirectUrl: options.redirectUrl ?? DEFAULT_REDIRECT_URL,
+    },
+  });
+
 describe("AuthCodeRedirectHandler", () => {
   describe("canHandle", () => {
     it("Accepts a valid url with the correct query", async () => {
       const authCodeRedirectHandler = getAuthCodeRedirectHandler();
-      expect(
-        await authCodeRedirectHandler.canHandle(
-          "https://coolparty.com/?code=someCode&state=oauth2_state_value"
-        )
-      ).toBe(true);
+      expect(await authCodeRedirectHandler.canHandle(mockRedirectUrl())).toBe(
```

Comment:
it may be better to use:
```suggestion
      expect(authCodeRedirectHandler.canHandle(mockRedirectUrl())).resolves.toBe(true)
```

In case the handler throws, that way you'll not bubble the exception / rethrow it via `await`.